docs/modules/ROOT/pages/explanations/decisions/postgres-monitoring.adoc

```diff
@@ -0,0 +1,109 @@
+= PostgreSQL Managed by VSHN Monitoring
+
+== Problem
+
+We need to provide PostgreSQL Managed by VSHN monitoring checks:
+
+* Availability check
+
+== Requirements for availability check
+
+* PostgreSQL is accessible on network level
+* it's possible to login to PostgreSQL
+* it's possible to execute queries 
+
+== Solutions
+
+PostgreSQL managed by Stackgres:
+
+* it doesn't reuse generated connection string which our users will use, which could lead to situation where our monitoring is green, but users can't login (issues with Service)
+* it's sidecar, issue with exporter would require pod restarting
+* prometheusAutobind from SGCluster doesn't work, so I have a fear that whole monitoring isn't really well tested
+
+PostgreSQL managed by VSHN:
+
+* the same functionalities as above
+* solves issues presented above
+
```

Comment:
I can see at least two further options that we should mention and at least explain why we don't want them:

* A pre-exiting query exporter that evaluates SQL queries in defined intervals. Something like: https://github.com/albertodonato/query-exporter (There are more that look similar)
  - That would be closer to measuring the actual user experience than whether the exporter works or not
  - Resolution could be a lot better as we could execute a query every 1-2 second instead of per scrape interval
* A custom SQL query prober.
   - Has same advantages as a pre-existing query prober
   - More flexibility. Could be helpful when we start to integrate all the exception into the SLA reporting
   - Downside is obviously more code we need to maintain